web3/middleware/filter.py

```diff
@@ -255,7 +255,7 @@ def __init__(
         if from_block is None or from_block == "latest":
             self._from_block = BlockNumber(w3.eth.block_number + 1)
         elif is_string(from_block) and is_hex(from_block):
-            self._from_block = BlockNumber(hex_to_integer(from_block))
+            self._from_block = BlockNumber(hex_to_integer(cast(HexStr, from_block)))
```

Comment:
seems like this type should be an `int` instead of casting to `HexStr`?

Reply:
I'm not sure I understand. In the line above, we confirm that it's a hex string, so we then want to convert it to an `int`. `hex_to_integer` is typed to accept only `HexStr`, so needs that casting. Maybe the answer is to open up typing on `hex_to_integer` to avoid these castings?

Reply:
oops, I misread. You're right. I thought we were casting the result of `hex_to_integer` to a `HexStr`. Don't mind me :)